code/modules/mob/living/carbon/xenomorph/castes/hunter/hunter.dm

```diff
@@ -30,3 +30,52 @@
 	SIGNAL_HANDLER
 	qdel(src)
 
+/mob/living/carbon/xenomorph/hunter/assassin
+	icon_state = "Assassin Hunter Running"
+	caste_base_type = /datum/xeno_caste/hunter/assassin
+
+
+/mob/living/carbon/xenomorph/hunter/assassin/change_form()
+	if(!loc_weeds_type)
+		to_chat(src, span_xenowarning("We need to be on weeds for us to shift again."))
+		return
+	wound_overlay.icon_state = "none"
+	if(do_after(src, 3 SECONDS, IGNORE_HELD_ITEM, src, BUSY_ICON_BAR, NONE, PROGRESS_GENERIC)) //dont move
+		do_change_form()
+
+///Finish the form changing of the hunter and give the needed stats
+/mob/living/carbon/xenomorph/hunter/assassin/proc/do_change_form()
+	playsound(get_turf(src), 'sound/effects/alien/new_larva.ogg', 25, 0, 1)
+	if(status_flags & INCORPOREAL)
+		var/turf/whereweat = get_turf(src)
+		if(whereweat.get_lumcount() == 0) //is it a lit turf
+			adjust_stagger(6 SECONDS) //think xenos get like half times on stuns due some multiplier, maybe less.
+			adjust_slowdown(6 SECONDS)
+		for(var/obj/machinery/light/lightie in range(rand(5,7), whereweat))
+			lightie.set_flicker(rand(3 SECONDS, 5 SECONDS), 3, 6)
```

Comment:
Why isn't this on the ability?

Reply:
I got the whole thing based on how hivemind and either forgot or thought there may be a reason it was done like that, either way i should move it yeah.